cli_tools/common/utils/daisyutils/traversal.go

```diff
@@ -0,0 +1,27 @@
+//  Copyright 2021 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package daisyutils
+
+import (
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+// To rebuild the mock, run `go generate ./...`
+//go:generate go run github.com/golang/mock/mockgen -package mocks -source $GOFILE -destination ../../../mocks/mock_workflow_traversal.go
+
+// WorkflowTraversal allows for modifying and validating a daisy workflow.
+type WorkflowTraversal interface {
+	Traverse(wf *daisy.Workflow) error
+}
```

Comment:
Most of the implementations are not really traversals, e.g. `ApplyAndValidateVars` and `ConfigureDaisyLogging` which is confusing.

`WorkflowModifier`, `WorkflowUpdater` or something similar would be a more appropriate name.

Reply:
sgtm -- Updated to `WorkflowModifier`.